File: bench/benches/tak.hs
import Bench

tak :: Integer -> Integer -> Integer -> Integer
tak x y z =
  if y < x
    then tak (tak (x - 1) y z) (tak (y - 1) z x) (tak (z - 1) x y)
    else z

main :: IO ()
main = bench "tak" (\u -> tak (with u 22) (with u 12) (with u 6))

File: bench/benches/tak.jl
include(joinpath(@__DIR__, "..", "lib", "bench.jl"))

tak(x, y, z) = y < x ? tak(tak(x - 1, y, z), tak(y - 1, z, x), tak(z - 1, x, y)) : z

bench("tak", () -> tak(22, 12, 6))
